### ISSUE TITLE
Batch AVSS protocol for double sharing at f<N/3

### ISSUE THREAD

**Issue by amiller** [initc3/HoneyBadgerMPC#162]
In general, a double sharing (a degree `2t` sharing) can only be robustly reconstructed if `t<N/4`, since we need `2t+1` correct points to interpolate, `3t+1` for robustness check, and need `N >= 4t+1` to tolerate `t` crashes. However, if we have an augmented sharing where the shares come along with their polynomial commitment, then we could validate the shares & reconstruction directly and would only require the `2t+1` points.

It is clear that we can AVSS double sharings at the `t<N/4` setting but based on the insight above can we modify the hbAVSS protocol to handle the `t<N/3` case? The challenge seems to be in the reconstruction phase (in hbAVSS light, we have to handle a "dealer implicate" case with only `t+1` nodes responding). We think we have a way around this by sending "double up" shares, so in AVSS each party receives `phi(i)` as well as `phi(N+i)`. This preserves privacy since the polynomial is `2t` rather than `t`, and it would enable reconstruction in the implicate phase. So, this issue is a placeholder in case we want to have a discussion about implementation options or in case it needs further thought.

**Comment by amiller** [comment]
This approach probably will not work as intended for MPC operations. While giving the adversary `2t` points out of a degree-`2t` polynomial still works for secret sharing, since it hides the secret, when used for composition the extra leakage can be a problem. Consider the following two functionalities:
`F_RandDouShaA` and `F_RandDouShaB`.

`F_RandDouShaA` does:
- Sample a random degree-`t` poly `r(.)`
- Sample a random degree-`2t` poly `s(.)` such that `s(0) = r(0)`
- Send each party `P_i` the share `r(i),s(i)`

`F_RandDouShaB` does:
- Sample a random degree-`t` poly `r(.)`
- Sample a random degree-`2t` poly `s(.)` such that `s(0) = r(0)`
- Send each party `P_i` the share `r(i),s(i),r(N+i),s(N+i)`

The two functionalities are different, and the protocol description in this first message may satisfy `F_RanDouShaB` but definitely not `F_RanDouShaA`. I'll make a separate note with an example of how the difference can matter for an MPC application.

For now, let me just describe an alternative protocol for realizing `F_RanDouShaB`.

`Prot_RanDouSha`:
- get a share of two degree-`t` polynomials `r(i),u(i)` from `F_Rand`, such that `r(0) = u(0)`. For here I'm assuming we use an instance of `hbAVSS` for sharing degree-`t` polynomials satisfying this constraint.
- get a share of an additional degree-`t` polynomial `v(i)` from `F_Rand`
- compute `s(i) = u(i) + v(i) i^t`
- output `r(i),s(i)`

Basically the idea is to generate a random degree `2t` polynomial by combining two degree-`t` polynomials `u(X)` and `v(X)`, by having `s(X) = u(X) + v(X) X^t`.
If `u` and `v` are uniform over `t` polynomials, then clearly `u + v X^t` is uniform over `2t` polynomials. Furthermore, the evaluation at `0` is conserved, `s(0) = u(0)`. Overall, the approach is similar. the parties get 2 shares during the protocol even though they output just 1, but the difference is in how they are combined.
